assets/src/components/BucketList.vue

```diff
@@ -32,51 +56,102 @@ export default {
       goals: [],
       activity: "",
       location: "",
-      error: null,
+      errors: [],
       uri: process.env.VUE_APP_API_URI
     }
   },
   created () {
     this.fetchGoals()
   },
   methods: {
+    fetchGoals: async function () {
+      this.goals = []
+
+      await axios.get(this.uri)
```

Comment:
I'm not very familiar with Vue.js so ignore if this makes no sense. But in a React app, I'd try to abstract away API calls into an API layer, just to separate out the responsibility of rendering from the responsibility of getting data from the backend. Something to consider for a future refactoring?

Reply:
Will keep this in mind when I work on injecting axios!